src/llvm.nim

```diff
@@ -29,24 +29,41 @@ type
 
 # Import the matching version:
 const
-  LLVMV* {. strdefine .} = "5.0"
+  LLVMV* {. strdefine .} = "12"
   LLVMLib* {. strdefine .} = "libLLVM-" & LLVMV & ".so"
   canDumpType* = false
 
+when parseFloat(LLVMV) >= 11.0:
+  declAll(
+    OpaqueModuleFlagEntry, OpaqueNamedMDNode, OpaqueValueMetadataEntry, OpaqueJITEventListener,
+    RemarkOpaqueString, RemarkOpaqueDebugLog, RemarkOpaqueArg, RemarkOpaqueEntry, RemarkOpaqueParser
+  )
+
+
 macro includeAll(version: static[string]): untyped =
   result = newNimNode(nnkStmtList)
 
-  for file in ["Types", "Support", "Core", "Target", "TargetMachine", "ExecutionEngine", "BitReader", "BitWriter", "IRReader", "Linker",
-               "Initialization", "Transforms/PassManagerBuilder"]:
-    result.add newTree(nnkIncludeStmt, newIdentNode($version & "/" & file))
+  when parseFloat(LLVMV) >= 11.0:
```

Comment:
nit: add a loop outside of the `when` statement for files that are imported in all versions.